scripts/fn_paradrop.sqf

```diff
@@ -0,0 +1,216 @@
+ * 2. Planes/helicopters in combatMode "COMBAT" have altitude overrides.
+ * 3. Drop distribution ultimately depends on AI pathing, and AI is stupid.
+ * 4. Function is designed to be called synchronously with craft reaching the
+ * WP from which it is called; if it is held up, then the drop WPs will not
+ * reflect the current azimuth.  If synchronizing WPs, recommend you sync WPs
+ * BEFORE the one that calls this function to avoid weird behavior, especially
+ * if the drop marker is close by.
+ *
+ * Optional Arguments: Drop Mode
+ * For convenience, this function has the following pre-set modes for drops:
+ * 0: Modern combat drop, 150m, parachutes deploy at 140m.
+ * 1: Dangerous combat drop, 105m, parachutes deploy at 80m. 
+ * 2: Training drop, 250m, parachutes deploy at 140m.
+ * 3: HALO drop, 4.5km, parachutes deploy at 900m.
+ * 4: Civilian drop, 3km, parachutes deploy at 900m. 
+ * For custom drops, drop altitude, drop zone length, deployment altitude can be set
+ * by providing an array of those values, in meters.
+ * 
+ * Note that for HALO or Civilian drops, most maps are not big enough to get to
+ * sufficient altitude and it is recommended to have several waypoints steadily
+ * increasing flyInHeight before using it.  Also, don't try with helicopters.
+ *
+ * Optional Arguments: Reset Height
+ * For drop purposes flyInHeight is set during the run.  If the transport is to be
+ * otherwise used after the run, set this to <true> in order to resume normal behavior
+ *
+ * Optional Arguments: Disable Waypoint 2
+ * Occasionally, AI behavior can be wonky especially when drop WPs are around airports.
+ * This will cause them to veer off from their drop direction and attempt to land. 
+ * Disabling WP 2 stops this behavior from occurring on the second waypoint.
+ * If it occurs on the first, I recommend changing the DZ area, the drop length,
+ * or the approach direction. 
+ *
+ * Arguments
+ * 0: Pilot Group <GROUP>
+ * 1: Drop Groups <ARRAY>
+ * 2: Drop Target <MARKER>
+ * 3: Drop Mode <ARRAY> or <NUMBER> (default: 0)
+ * 4: Reset Height <BOOL> (default: false) 
+ * 5: Disable Waypoint 2 <BOOL> (default: false)
+ *
+ * Return Value:
+ * None
+ *
+ * Examples:
+ * [pilot_group, [airborne_1], "drop_marker"] call TAC_Scripts_fnc_paradrop;
+ * [pilot_group, [airborne_1,airborne_2], "drop_marker"] call TAC_Scripts_fnc_paradrop;
+ * [pilot_group, [airborne_1], "drop_marker", 1] call TAC_Scripts_fnc_paradrop;
+ * [pilot_group, [airborne_1], "drop_marker", 0, false, false] call TAC_Scripts_fnc_paradrop;
+ * [pilot_group, [airborne_1], "drop_marker", [200,1000,150]] call TAC_Scripts_fnc_paradrop;
+ */
+
+#define DROP_MODES [[150, 1000, 140], [105, 1000, 80], [250, 1000, 140] ,[4500, 1000, 900], [3000, 1000, 900]]
+
+params ["_pilotGroup", "_dropGroups", "_loc", ["_dropMode", 0], ["_resetHeight", false], ["_disableWP2", false]];
+
+if (_loc isEqualTo "") exitWith {ERROR("Paradrop given invalid marker!")};
+
```

Comment:
Incorporating the fix for a bug you mentioned in previous review.
```suggestion

// make sure you delete these 2 lines later below in the script (and this comment too)! I just moved them here as early exit is better, and we have a use for it early here now
private _vehicle = vehicle (leader _pilotGroup);
if (isNull _vehicle) exitWith {ERROR("Paradrop pilot group not in vehicle!")};

private _origDropGroupsCount = count _dropGroups;
_dropGroups = _dropGroups select {_vehicle != vehicle (leader _x)};
if (count _dropGroups < _origDropGroupsCount) then {
    WARNING_1("Paradrop %1 drop group(s) skipped because they are not in the same vehicle as pilot group!",_origDropGroupsCount - count _dropGroup)
};
```